bcipy/signal/model/offline_analysis.py

```diff
@@ -333,115 +338,125 @@ def analyze_gaze(
         labels = np.array([sym] * len(le))  # Labels are the same for both eyes
         train_le, test_le, train_labels_le, test_labels_le = subset_data(le, labels, test_size=0.2, swap_axes=False)
         train_re, test_re, train_labels_re, test_labels_re = subset_data(re, labels, test_size=0.2, swap_axes=False)
+        train_dict[sym] = np.concatenate((train_le, train_re), axis=0)
         test_dict[sym] = np.concatenate((test_le, test_re), axis=0)
 
         # Fit the model based on model type.
-        # Model 1: Fit Gaussian mixture (comp=2) on each symbol and each eye separately
         if model_type == "Individual":
-            model.fit(train_re)
+            # Model 1: Fit Gaussian mixture on each symbol separately
+            model.fit(train_dict[sym])
 
-            means, covs = model.evaluate(test_re)
+            means, covs = model.evaluate()
 
             # Visualize the results:
             figure_handles = visualize_gaze_inquiries(
                 le, re,
                 means, covs,
                 save_path=None,
-                show=show_figures,
+                show=False,
                 raw_plot=True,
             )
-            figures.extend(figure_handles)
+            figures.append(figure_handles)
             left_eye_all.append(le)
             right_eye_all.append(re)
             means_all.append(means)
             covs_all.append(covs)
 
-            # TODO: Calculate scores for the test set.
-            # scores, predictions = model.predict(test_re)
-
-        # Model 2: Fit Gaussian mixture (comp=1) on a centralized data
         if model_type == "Centralized":
             # Centralize the data using symbol positions:
             # Load json file.
             with open(f"{BCIPY_ROOT}/parameters/symbol_positions.json", 'r') as params_file:
                 symbol_positions = json.load(params_file)
             # Subtract the symbol positions from the data:
-            centralized_data_left.append(model.reshaper.centralize_all_data(train_le, symbol_positions[sym]))
-            centralized_data_right.append(model.reshaper.centralize_all_data(train_re, symbol_positions[sym]))
+            centralized_data_train.append(model.reshaper.centralize_all_data(train_dict[sym], symbol_positions[sym]))
 
     if model_type == "Individual":
-        # Takes in all means and covs from individual symbols, and
-        # calculates the likelihoods and predictions for each test point.
-        # Convert test_dict to a list of arrays:
         accuracy = 0
         acc_all_symbols = {}
         counter = 0
         for sym in symbol_set:
-            # Skip if test_dict is empty for certain symbols:
+            # Continue if there is no test data for this symbol:
             if len(test_dict[sym]) == 0:
                 acc_all_symbols[sym] = 0
                 continue
```

Comment:
I don't think this continue is needed...

Reply:
It actually crashes without it :(